bpfd-operator/apis/v1alpha1/shared_types.go

```diff
@@ -237,6 +237,9 @@ const (
 	// BpfProgCondMapOwnerNotLoaded indicates that the eBPF program sharing a map with another
 	// eBPF program and that program is not loaded.
 	BpfProgCondMapOwnerNotLoaded BpfProgramConditionType = "MapOwnerNotLoaded"
+	// BpfProgCondByteCodeError indicates that an error occured when trying to
```

Comment:
There's only a few failure modes here right? i.e something along the lines of 

1. Failure to pull the image (various errors)
2. Failure to find the file specified at PATH

Should we have two explicit condition types for this?

Reply:
BTW, I think it would be good to be able to report these errors as well, but I think those are different failures and could be handled in another pr.  If you agree, I can open an issue for them.

Reply:
https://github.com/bpfd-dev/bpfd/issues/659